src/utils/testDatasource.ts

```diff
@@ -0,0 +1,39 @@
+import { HealthCheckResult } from '@grafana/runtime';
+
+// These are from core Grafana - they are not exported
+export type HealthCheckResultDetails = Record<string, any> | undefined
+export class HealthCheckError extends Error {
+  details: HealthCheckResultDetails;
+
+  constructor(message: string, details: HealthCheckResultDetails) {
+    super(message);
+    this.details = details;
+    this.name = 'HealthCheckError';
+  }
+}
+
+export type TestDatasourceReturn = Partial<HealthCheckResult> | Error
+export type BaseTestDatasource = () => Promise<TestDatasourceReturn>
+
+/**
+ * Override function for testDatasource in Grafana for the health check
+ * 
+ * There is a new error format that is introduced in Grafana 8 which is not backwards compatible by default
+ * 
+ * This transforms the health check result into the old format if Grafana <8 is being used
+ *
+ * @param baseTestDatasource The original testDatasource function
+ * @returns Either the health check result if the health check was successful or an error that is handled later by Grafana
+ */
+export const testDatasource = async (baseTestDatasource: BaseTestDatasource): Promise<TestDatasourceReturn> => {
```

Comment:
checkDatasource?  testDatasource is confusing to me, I assume its a unit test

Reply:
this is the function we want to override the core Grafana's `testDatasource` with: https://github.com/grafana/grafana/blob/master/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts#L207

I was thinking it would be good to have the same name?

Reply:
yeah that makes sense, just confusing

Reply:
Yeah, I can see how that can be confusing - hopefully the jsdocs I've added can help clarify things